doc/vxpwlfmt.8
.TH "vxpwlfmt" "8" "2008-01-05" "Vitalnix" "Vitalnix User Management Suite"
.SH "Name"
.PP
vxpwlfmt \(em password list pretty-formatting
.SH "Syntax"
.PP
\fBvxpwlfmt\fP [\fB\-w\fP] [\fB\-t\fP \fIfile\fP] \fB\-i\fP \fIfile\fP
\fB\-o\fP \fIfile\fP \fB\-s\fP \fIname\fP
.SH "Description"
.PP
When \fBvxusersync\fP(8) adds and deletes users, it notes down their usernames
and, for new users, their passwords in a simple textfile. \fBvxpwlfmt\fP will
take such a logfile and "pretty print" it in a more advanced format such as
HTML or RTF, suitable for distribution. Each "style", as it is called, outputs
the data in a different layout and possibly different format.
.SH "Options"
.TP
\fB\-i\fP \fIfile\fP
Input file, as generated by libvxmdsync.
.TP
\fB\-o\fP \fIfile\fP
Place output in \fIfile\fP.
.TP
\fB\-s\fP \fIname\fP
Use the specified style when converting the logfile to "readable" output.
.TP
\fB\-t\fP \fIfile\fP
Use the specified template file for output. This option is only required one
styles that require one.
.TP
\fB\-w\fP
Instead of converting, show all available styles. Those that require a template
will be marked appropriately.
.PP
The \-i, \-o and \-s options are mandatory.
.SH "Available styles"
.SS "a1"
.PP
Prints exactly one table containing all users, in text/plain.
.SS "pg_html"
.PP
Generates one table per private group descriptor and outputs in text/html.
.SS "pg_rtf"
.PP
Generates one table per private group descriptor, with a page break after each
table. The output format is text/rtf, which can be read by e.g. Microsoft
Office. This style requires a template file; there are two templates shipped,
share/sg.rtf (English) and share/sg_DE.rtf (German).
.SS "pg_txt"
.PP
Generates one table per private group descriptor, with a page break after each
table (by ASCII fashion this is the form-feed character, '\ef'). The output
format is text/plain, and line endings are what the system outputs for '\en'.
Under UNIX this means it is just '\en' and you will need an extra preprocessing
step to convert this to '\er\en' if sending it directly to the printer over
/dev/lp*.
.SS "sb"
.PP
Generates a nice form for each user where his/her login details and possibly
other info is noted. The output is a TeX document.
.PP
The purpose of this style is that each user gets a letter, possibly wrapped
into an envelope to not unnecessarily reveal the password or other sensitive
information.
.SH "Examples"
.PP
Assuming \fBvxusersync\fP created a logfile called run0919.log, and you want to
have a nice ASCII page of all new users:
.PP
mdpwlfmt \-i run0919.log \-o run0919.txt \-s pg_txt
.PP
You may then use \fBlpr\fP(1) or other means to print it, if you wish so. Of
course, it also nice to look at on-screen.
.PP
The "pg_rtf" (as well as "pg_html") style is especially important, because it
allows you to change the layout a little bit more, depending on how you would
like to have it. pg_rtf uses a (quite complex) pre-created RTF file for its
output, and as such, requires a the template option \-t:
.PP
mdpwlfmt \-i run0919.log \-o run0919.rtf \-s pg_rtf \-t
/opt/vitalnix/share/vitalnix/pg.rtf
.PP
The resulting file may then be opened using a word processor (most likely
Microsoft Word) and adjusted as needed.
.SH "See also"
.PP
\fBvitalnix\fP(7)


src/doc/vxpwlfmt.8.php
<?php include_once("Base-header.php"); ?>

<h1>Name</h1>

<p>vxpwlfmt&nbsp;-- password list formatter</p>

<h1>Syntax</h1>

<p class="code"><code><b>vxpwlfmt</b> [<b>-w</b>] [<b>-t</b> <i>file</i>]
<b>-i</b> <i>file</i> <b>-o</b> <i>file</i> <b>-s</b> <i>name</i></code></p>

<h1>Description</h1>

<p class="block">When <i>vxusersync</i> adds and deletes users, it notes down
their usernames and, for new users, their passwords in a simple textfile.
<i>vxpwlfmt</i> will take such a logfile and "pretty print" it in a more
advanced format such as HTML or RTF, suitable for distribution. Each "style",
as it is called, outputs the data in a different layout and possibly different
format.</p>

<h1>Options</h1>

<table border="1">
	<tr>
		<td class="t1n"><code><b>-i</b> <i>file</i></code></td>
		<td class="t1">Input file, as generated by <i>libvxmdsync</i></td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-o</b> <i>file</i></code></td>
		<td class="t2">Place output in <i>file</i></td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-s</b> <i>name</i></code></td>
		<td class="t1">Use the specified style when converting the
		logfile to "readable" output</td>
	</tr>
	<tr>
		<td class="t2n"><code><b>-t</b> <i>file</i></code></td>
		<td class="t2">Use the specified template file for output. This
		option is only required one styles that require one.</td>
	</tr>
	<tr>
		<td class="t1n"><code><b>-w</b></code></td>
		<td class="t1">Instead of converting, show all available
		styles. Those that require a template will be marked
		appropriately.</td>
	</tr>
</table>

<p class="block">The <code>-i</code>, <code>-o</code> and <code>-s</code>
options are mandatory.</p>

<h1>Available styles</h1>

<h2>a1</h2>

<p class="block">Prints exactly one table containing all users, in
text/plain.</p>

<h2>pg_html</h2>

<p class="block">Generates one table per private group descriptor and outputs
in text/html.</p>

<h2>pg_rtf</h2>

<p class="block">Generates one table per private group descriptor, with a page
break after each table. The output format is text/rtf, which can be read by
e.g. Microsoft Office. This style requires a template file; there are two
templates shipped, <code>share/sg.rtf</code> (English) and
<code>share/sg_DE.rtf</code> (German).</p>

<h2>pg_txt</h2>

<p class="block">Generates one table per private group descriptor, with a page
break after each table (by ASCII fashion this is the form-feed character,
<code>'\f'</code>). The output format is text/plain, and line endings are what
the system outputs for <code>'\n'</code>. Under UNIX this means it is just
<code>'\n'</code> and you will need an extra preprocessing step to convert this
to <code>'\r\n'</code> if sending it directly to the printer over
<code>/dev/lp*</code>.</p>

<h2>sb</h2>

<p class="block">Generates a nice form for each user where his/her login
details and possibly other info is noted. The output is a TeX document.</p>

<p class="block">The purpose of this style is that each user gets a letter,
possibly wrapped into an envelope to not unnecessarily reveal the password or
other sensitive information.</p>

<h1>Examples</h1>

<p class="block">Assuming <i>vxusersync</i> created a logfile called
<code>run0919.log</code>, and you want to have a nice ASCII page of all new
users:</p>

<p class="code"><code>mdpwlfmt -i run0919.log -o run0919.txt -s
pg_txt</code></p>

<p class="block">You may then use <i>lpr</i> or other means to print it, if you
wish so. Of course, it also nice to look at on-screen.</p>

<p class="block">The <i>pg_rtf</i> (as well as <i>pg_html</i>) style is
especially important, because it allows you to change the layout a little bit
more, depending on how you would like to have it. <i>pg_rtf</i> uses a (quite
complex) pre-created RTF file for its output, and as such, requires a the
template option <code>-t</code>:</p>

<p class="code"><code>mdpwlfmt -i run0919.log -o run0919.rtf -s pg_rtf -t
/opt/vitalnix/share/vitalnix/pg.rtf</code></p>

<p class="block">The resulting file may then be opened using a word processor
(most likely Microsoft Word) and adjusted as needed.</p>

<h1>List of styles</h1>

<p class="block">A brief description of what styles are available and what they
are intended for can be found <a href="extra-pwlstyles.php">List of PWL styles
page</a>.</p>

<h1>See also</h1>

<p><a href="vitalnix.7.php">vitalnix</a>(7)</p>

<?php include_once("Base-footer.php"); ?>
